qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -113,16 +113,16 @@ def models(self) -> List[lmfit.Model]:
         """Return fit models."""
 
     @property
-    def drawer(self) -> BaseCurveDrawer:
-        """A short-cut for curve drawer instance."""
-        return self._options.curve_drawer
+    def plotter(self) -> BasePlotter:
```

Comment:
Plotters and drawers are different in that you plot a figure but draw lines or shapes on a canvas. Given that this PR uses plotters as a bridge/intermediary between analyses and drawing on a canvas, it makes sense that the analysis class contains the plotter, which intern contains the drawer. With the new classes, and their names, the analysis classes need to access the plotter instead of the drawer, hence the name-change. If `.drawer` needs to be kept for deprecation purposes, I can make it return `.plotter.drawer` until it's removed in the future.

Reply:
Partially addressed in 7101687. I intend to add a new `BaseDrawer` subclass to wrap deprecated `BaseCurveDrawer` instances. This should allow users to run old code during the deprecation period.

Reply:
Addressed fully in 28a8c80.